orderer/consensus/etcdraft/chain.go

```diff
@@ -1115,14 +1115,28 @@ func (c *Chain) commitBlock(block *common.Block) {
 func (c *Chain) detectConfChange(block *common.Block) *MembershipChanges {
 	// If config is targeting THIS channel, inspect consenter set and
 	// propose raft ConfChange if it adds/removes node.
+	c.logger.Infof("Detected configuration change")
+
+	// if the consensusType is bft, then configMetadata which represents the raft metadata should be nil
 	configMetadata, consensusType := c.newConfigMetadata(block)
+	c.logger.Infof("Detected configuration change: consensusType is: %s, configMetadata is: %v", consensusType, configMetadata)
 
-	if configMetadata == nil {
+	if consensusType == nil {
+		c.logger.Infof("ConsensusType is %v", consensusType)
 		return nil
 	}
 
 	if consensusType.Type != "etcdraft" {
-		c.logger.Infof("Detected migration to %s", consensusType.Type)
+		if consensusType.Type == "BFT" {
+			c.logger.Infof("Detected migration to %s", consensusType.Type)
+			return nil
+		} else {
+			c.logger.Panicf("illegal consensus type detected: %s", consensusType.Type)
```

Comment:
this is post consensus. Do we have a validation check for this condition pre-consensus anywhere?

Reply:
otherwise, you know what will happen once we reach this point ;-)

Reply:
We have one more check in the maintenance filter.

Reply:
Yes, May is correct, we prevent it the maintenance filter.